### ISSUE TITLE
Cell.Value is corrupted with furigana

### ISSUE THREAD

**Issue by udaken** [rabanti-github/NanoXLSX#25]
An xlsx file created with the Japanese version of Excel 2019 contains an annotation called [Furigana](https://en.wikipedia.org/wiki/Furigana).
When that xlsx file is read in the NanoXLSX file, the furigana is mixed in the Cell.Value, resulting in an incorrect value.

Cell.Value is expected to ignore the furigana.

Steps to reproduce the problem:
1. use [tokyo.xlsx](https://github.com/rabanti-github/NanoXLSX/files/8369616/tokyo.xlsx)
   <img src="https://user-images.githubusercontent.com/7896698/160571091-7431c8c4-dc10-4278-9af4-738e51d0bf17.png">
2. Execute the following code.
   ```cs
    using NanoXLSX;
    var wb = Workbook.Load("tokyo.xlsx");
    Console.Write.WriteLine(wb.Worksheets[0].Cells["A1"].Value);
    
    // expect value is "ここは東京"
    // but, print "ここは東京トウキョウ"
    ```

**Comment by udaken** [comment]
My guess is that we can just ignore the `rPH` tag.
<img src="https://user-images.githubusercontent.com/7896698/160577040-2fdd7182-8420-4eca-aced-d56d58eb8c8d.png">

**Comment by rabanti-github** [comment]
こんいちは

Thank you for the report.

So, have I understood the issue correct:
The Furigana part above the Kana should never be visible in Excel. But it should be preserved when you copy the value of a cell to a program that handle Furigana?
I am asking because, when we ignore, rPh, the Furigana signs will be lost.

I will have a look into the formatting in the shared string table. This (inline formatting) is on top of my backlog as ToDo.

Thank you in advance for clarification.

**Comment by udaken** [comment]
Yes, Excel does not display the furigana part.

I have been using files containing Japanese for over 10 years and have never needed furigana.
Even when I edit files in Excel for the web, the furigana is lost.

Therefore, my conclusion is that there is no problem with lost furigana.

For a program to preserve the furigana, you generally need a different object model than a single value.
VBA provides [Characters.PhoneticCharacters](https://docs.microsoft.com/en-us/office/vba/api/excel.characters.phoneticcharacters).

**Comment by rabanti-github** [comment]
OK, thank you.

When I implemented the stared string table, I decided to remove all paragraphs to preserve any content. However, since Furigana is only a Kanji transcription, it's presence breaks the sentence / word structure ,as you already stated.

Let me look for the best solution to preserve formatted text and to remove transcriptions. I have analyze the rPH (and possible others) tag a little bit. Maybe it can always be ignored or there will be an import option, if there are use cases, where rPH content is meaningful.

Please give me some time to do this. Currently, I am working on a big refactoring (dev branch) and I have to adapt any change on several braches and projects.

どうもありがとうございました。

**Comment by rabanti-github** [comment]
I have some good news.
The rPH tag seems to be designated only for transcriptions (as Furigana is) of Han-originated characters. So it can be removed from the original text without side-effects.
I will do this by default with an opt-in import-option to keep the transcriptions, enclosed in brackets, if wished.
Please give me some days to implement and test this feature.

**Comment by rabanti-github** [comment]
Hello
Technically, I already have a solution.
However. I have insufficient test data. And it looks like my western Office refuses to carry Furigana at all. I tried to copy-paste from a web browser and Word, but Excel never created a rPH tag.

Could you provide me with some additional samples, where Kanji characters are transcribed by Furigana? I would suggest something like that:

- `お元気ですか` --> Two connected Kanji surrounded by Hiragana
- `この所は賑やかですぞ。`--> Two disconnected Kanji, surrounded by Hiragana
- `金` --> One single Kanji
- `富士山は高いです。`--> Several Kanji at the beginning of the text and Hiragana at the end
- `この儘`--> Hiragana with Kanji at the end of the text

If you could provide me an Excel file with these (or similar) use cases, I could test my patch properly. Currently, I cannot predict the right tag structure.

Thank you very much in advance.

**Comment by udaken** [comment]
Thank you for your quick response.

We will gladly provide sample data. Please wait a moment.

**Comment by udaken** [comment]
Sample data attached.
I hope this will be useful to you.

[furigana.xlsx](https://github.com/rabanti-github/NanoXLSX/files/8393700/furigana.xlsx)

**Comment by rabanti-github** [comment]
ただいま
Thank you for the samples. They helped me a lot.
I just released v1.8.6. The Nuget packet will be available in some minutes.
by default, Furigana will now be discarded. If the transcription is wished, there is an import option "EnforcePhoneticCharacterImport", that can be set to true. In this case, Furigana will be appended to the Kanji characters:
|Default   | With Import Option  |
|---|---|
| お元気ですか  | お元気(ゲンキ)ですか   |
| この所は賑やかです   | この所(トコロ)は賑(ニギ)  |
| この所は賑やかです  | この所(トコロ)は賑(ニギ)やかです   |
| 金   | 金(カネ)  |
|  富士山は高いです。| 富士山(フジサン)は高(タカ)いです。 |
| この儘 | この儘(ママ) |

Please let me know if something is not working as expected.
また、レポートありがとうございました。

**Comment by rabanti-github** [comment]
I will close this issue for now. If you try out the fix and find something that is not as you would expecting, please do not hesitate to re-open or file a new issue.
Thank you
